packages/core/src/utils/nuxt/_proxyUtils.ts

```diff
@@ -38,7 +38,12 @@ export const getIntegrationConfig = (context: NuxtContext, configuration: any) =
       baseURL: new URL(/\/api\//gi.test(baseURL) ? '' : 'api', baseURL).toString(),
       headers: {
         ...(cookie ? { cookie } : {}),
-        ...(context.req ? { Host: context.req.headers.host } : {})
+        ...(context.req ? {
```

Comment:
These ternaries with destructuring terrify me. I think we could make it more readable... It's hard to tell what is going on there at first sight.

Approved, but please, consider my comment.

Reply:
Could we do it like this instead?

```typescript
 const resolvedHost = context?.req?.headers?.['x-forwarded-host'] || context?.req?.headers?.host;

  return merge({
    axios: {
      baseURL: new URL(/\/api\//gi.test(baseURL) ? '' : 'api', baseURL).toString(),
      headers: {
        ...(cookie ? { cookie } : {}),
        ...(resolvedHost ? { Host: resolvedHost } : {})
      }
    }
  }, configuration);
```

And as I received comment from @WojtekTheWebDev in [PR#677](https://github.com/vuestorefront/bigcommerce/pull/677) we don't need 'x-forwarded-host' and needed to ensure Host is there coming from either x-forwarded-host or directly from host property of request headers.

Reply:
Well @jeffpdotone, I don't really like the idea of merging two headers into one. If we have 2 headers let's leave them as they are during forwarding or we will end up having confusing code. To be honest, that was the primary goal of this PR.

Reply:
All good then if the purpose is served and fixed the main issue.